atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/SqliteOracleAdapter.java

```diff
@@ -0,0 +1,309 @@
+                unusedConfig);
+        tableInitializer.createMetadataTable(METADATA_TABLE.getQualifiedName());
+    }
+
+    public TableDetails createTable(String prefix, String tableName, String owner) {
+        // This is so that unique table metadata is created for different prefixes and owners
+        // While Oracle doesn't handle the first case (it creates tables then throws when adding metadata!), tables
+        // are scoped to owners. The below attempts to mimic that.
+        Namespace namespace = Namespace.create(prefix + owner);
+        TableReference tableReference = TableReference.create(namespace, tableName);
+
+        // Equivalent to prefixed table name. Since we want control over
+        String reversibleTableName = generatePrefixedTableName(prefix, tableReference);
+
+        // We don't actually shrink it, but we do add a unique identifier to mimic the counting that ddltable does
+        String physicalTableName =
+                reversibleTableName + UUID.randomUUID().toString().replace("-", "");
+
+        // We don't use ddlTable createTable since we want the table name to insert into allTables, and extracting
+        // that is quite painful
+        return runWithConnection(connection -> {
+            PreparedStatement insertAllTables =
+                    connection.prepareStatement("INSERT INTO all_tables VALUES (upper(?), " + "upper(?))");
+            insertAllTables.setString(1, physicalTableName);
+            insertAllTables.setString(2, owner);
+            insertAllTables.executeUpdate();
+
+            String createTableSql = "CREATE TABLE %s (k VARCHAR(128) PRIMARY KEY, value VARCHAR(32))";
+            Statement createNewTable = connection.createStatement();
+            createNewTable.executeUpdate(String.format(createTableSql, physicalTableName));
+
+            PreparedStatement insertMetadata = connection.prepareStatement(
+                    "INSERT INTO " + METADATA_TABLE.getQualifiedName() + " (table_name, table_size) VALUES (?, ?)");
+            insertMetadata.setString(1, tableReference.getQualifiedName());
+            insertMetadata.setInt(2, 1);
+            insertMetadata.executeUpdate();
+
+            PreparedStatement insertTableMapping = connection.prepareStatement("INSERT INTO "
+                    + AtlasDbConstants.ORACLE_NAME_MAPPING_TABLE + " (table_name, short_table_name) VALUES (?, ?)");
+            insertTableMapping.setString(1, reversibleTableName);
+            insertTableMapping.setString(2, physicalTableName);
+            insertTableMapping.executeUpdate();
+            return TableDetails.builder()
+                    .tableReference(tableReference)
+                    .physicalTableName(physicalTableName)
+                    .reversibleTableName(reversibleTableName)
+                    .build();
+        });
+    }
+
+    public Set<String> listAllTables() {
```

Comment:
We might want to distinguish SQLite tables from Atlas-level tables in some way - don't have a great solution, other than maybe wrapping the Atlas ones in TableReference (see comment on `listAllTablesWithMapping`)

Reply:
I already filter out all the sqlite tables via \AND name NOT LIKE 'sqlite_%'\, unless I'm missing something that search and testing didn't show.

Reply:
discussed offline: I meant the SQLite names vs the AtlasDB names of Atlas data tables (you rightly filter out all the SQLite ones)